domain/src/main/java/org/oppia/android/domain/topic/TopicListController.kt

```diff
@@ -261,37 +346,360 @@ class TopicListController @Inject constructor(
+        recentlyPlayerChapterSummary.explorationId,
+        isTopicConsideredCompleted
+      )
     }
-    val totalChapterCount = storyData
-      .getJSONObject(0)
-      .getJSONArray("node_titles")
-      .length()
-    val storyId = storyData.optJSONObject(0).optString("id")
-    val storySummary = topicController.retrieveStory(topicId, storyId)
+    return null
+  }
 
-    val promotedStoryBuilder = PromotedStory.newBuilder()
-      .setStoryId(storyId)
-      .setStoryName(storySummary.storyName)
-      .setLessonThumbnail(storySummary.storyThumbnail)
-      .setTopicId(topicId)
-      .setTopicName(topicJson.optString("topic_name"))
-      .setCompletedChapterCount(0)
-      .setTotalChapterCount(totalChapterCount)
-    if (storySummary.chapterList.isNotEmpty()) {
-      promotedStoryBuilder.nextChapterName = storySummary.chapterList[0].name
-      promotedStoryBuilder.explorationId = storySummary.chapterList[0].explorationId
+  private fun createOngoingStoryListBasedOnMostRecentlyCompleted(
+    storyId: String,
+    story: StorySummary,
+    latestCompletedChapterProgress: ChapterProgress,
+    completedChapterProgressList: List<ChapterProgress>,
+    topic: Topic,
+    isTopicConsideredCompleted: Boolean
+  ): PromotedStory? {
+    val lastChapterSummary: ChapterSummary? =
+      story.chapterList.find { chapterSummary ->
+        latestCompletedChapterProgress.explorationId == chapterSummary.explorationId
+      }
+    val nextChapterIndex = story.chapterList.indexOf(lastChapterSummary) + 1
+    if (story.chapterList.size > nextChapterIndex) {
+      val nextChapterSummary: ChapterSummary? = story.chapterList[nextChapterIndex]
+      if (nextChapterSummary != null) {
+        return createPromotedStory(
+          storyId,
+          topic,
+          completedChapterProgressList.size,
+          story.chapterCount,
+          nextChapterSummary.name,
+          nextChapterSummary.explorationId,
+          isTopicConsideredCompleted
+        )
+      }
+    }
+    return null
+  }
+
+  private fun ChapterProgress.getNumberOfDaysPassed(): Long {
+    return TimeUnit.MILLISECONDS.toDays(
+      oppiaClock.getCurrentCalendar().timeInMillis - this.lastPlayedTimestamp
+    )
+  }
+
+  // TODO(#2550): Remove hardcoded order of topics. Compute list of suggested stories from backend structures
+  /** Returns a list of topic IDs for which the specified topic ID expects to be completed before
+   * being suggested.
+   * */
+  private fun retrieveTopicDependencies(topicId: String): List<String> {
+    return when (topicId) {
+      // TEST_TOPIC_ID_0 (depends on Fractions)
+      TEST_TOPIC_ID_0 -> listOf(FRACTIONS_TOPIC_ID)
+      // TEST_TOPIC_ID_1 (depends on TEST_TOPIC_ID_0,Ratios)
+      TEST_TOPIC_ID_1 -> listOf(TEST_TOPIC_ID_0, RATIOS_TOPIC_ID)
+      // Fractions (depends on A+S, Multiplication, Division)
+      FRACTIONS_TOPIC_ID -> {
+        listOf(ADDITION_AND_SUBTRACTION_TOPIC_ID, MULTIPLICATION_TOPIC_ID, DIVISION_TOPIC_ID)
```

Comment:
We need to remove all topics from this method (code-wise) which aren't yet actually in the app (so everything except test topics + fractions/ratios). Please add a note above the ``when`` that the comments describe the correct dependencies, but those might not be available until the topic is introduced into the app.

Reply:
Done.